subprojects/shipkit/src/test/groovy/org/shipkit/internal/util/RetryManagerTest.groovy

```diff
@@ -0,0 +1,53 @@
+package org.shipkit.internal.util
+
+import spock.lang.Specification
+
+class RetryManagerTest extends Specification {
+
+    def "should retry when retries limit not reached"() {
+        given:
+        RetryManager retryCounter = RetryManager.defaultRetryValues()
+
+        expect:
+        retryCounter.shouldRetry()
+    }
+
+    def "should not retry when retries limit reached"() {
+        given:
+        RetryManager retryCounter = RetryManager.defaultRetryValues()
+        for (int i = 0; i < 20; i++) {
+            retryCounter.waitNow({ })
+        }
+
+        expect:
+        !retryCounter.shouldRetry()
+    }
+
+    def "providing correct description"() {
+        given:
+        RetryManager retryCounter = RetryManager.defaultRetryValues()
+        for (int i = 0; i < 7; i++) {
+            retryCounter.waitNow({ })
+        }
+
+        expect:
+        retryCounter.describe() == "Waiting time so far: 210 seconds. Waiting 70 seconds..."
```

Comment:
Does it mean that this test executes in minutes? Let's make sure the unit tests are supersonic fast so that we get the best dev productivity. I suggest to create manager with super short timeouts for the purposes of unit testing.

Reply:
No it does not. 

Waiting method is delivered by lambda expression. For tests it's an empty lambda so every wait method just does nothing and is fast.

Reply:
Great!